src/libraries/System.Net.WebSockets.Client/src/System/Net/WebSockets/BrowserWebSockets/BrowserWebSocket.cs

```diff
@@ -476,13 +476,17 @@ private async Task SendAsyncCore(ArraySegment<byte> buffer, WebSocketMessageType
             try
             {
                 var sendTask = BrowserInterop.UnsafeSendSync(_innerWebSocket!, buffer, messageType, endOfMessage);
-                if (sendTask == null)
+                if (sendTask != null)
                 {
-                    // return synchronously
-                    return;
+                    await CancelationHelper(sendTask, cancellationToken, FastState).ConfigureAwait(true);
                 }
-
-                await CancelationHelper(sendTask, cancellationToken, FastState).ConfigureAwait(true);
+#if FEATURE_WASM_THREADS
+                // return synchronously, not supported with MT
+                else
+                {
+                    Environment.FailFast("Unexpected synchronous result");
```

Comment:
A more detailed message here would be good, though FailFast messages tend to not flow through in the browser anyway

Reply:
I think we fixed the abort message flow recently.